### PR TITLE
feat(docker): create volume req

### LINKED ISSUE #278: [Docker] Create volume request
Create the structure to receive the `CreateVolumeRequest` and publish the `AvailableVolumes` property.

<details><summary>Interfaces</summary>
<p>

### Create Volume Request

```json
{
  "interface_name": "io.edgehog.devicemanager.apps.CreateVolumeRequest",
  "version_major": 0,
  "version_minor": 1,
  "type": "datastream",
  "ownership": "server",
  "aggregation": "object",
  "mappings": [
    {
      "endpoint": "/volume/id",
      "type": "string"
    },
    {
      "endpoint": "/volume/driver",
      "type": "string"
    },
    {
      "endpoint": "/volume/options",
      "type": "stringarray",
      "doc": "An array of key=value options to set for the drive"
    }
  ]
}
```

### Available Volumes

```json
{
  "interface_name": "io.edgehog.devicemanager.apps.AvailableVolumes",
  "version_major": 0,
  "version_minor": 1,
  "type": "properties",
  "ownership": "device",
  "mappings": [
    {
      "allow_unset": true,
      "endpoint": "/%{volume_id}/created",
      "type": "boolean"
    },
    {
      "allow_unset": true,
      "endpoint": "/%{volume_id}/driver",
      "type": "string"
    },
    {
      "allow_unset": true,
      "endpoint": "/%{volume_id}/options",
      "type": "stringarray",
      "doc": "An array of key=value options to set for the drive"
    }
  ]
}
```

</p>
</details>

### PR DESCRIPTION
**Pull request by joshuachp**
Closes #278

### CODE REVIEW

File: edgehog-device-runtime-containers/src/docker/volume.rs [hunk 1]
```diff
@@ -0,0 +1,336 @@
...
+                warn!("volume not found: {message}");
+
+                return Ok(None);
+            }
+            Err(err) => return Err(VolumeError::Inspect(err)),
+        };
+
+        trace!("volume info: {volume:?}");
+
+        Ok(Some(volume))
```
[the remark this answers is not in the document]
Comment (joshuachp) [reply edgehog-device-runtime-containers/src/docker/volume.rs:121]: Yeh, that's better

File: edgehog-device-runtime-containers/src/docker/volume.rs [hunk 2]
```diff
@@ -0,0 +1,336 @@
...
+
+                Ok(Some(volume))
+            }
+            Err(BollardError::DockerResponseServerError {
+                status_code: 404,
+                message,
+            }) => {
+                warn!("volume not found: {message}");
+
+                return Ok(None);
```
Comment (harlem88) [inline edgehog-device-runtime-containers/src/docker/volume.rs:118]: You can remove `return`.
```suggestion
                Ok(None)
```

File: edgehog-device-runtime-containers/src/docker/volume.rs [hunk 3]
```diff
@@ -0,0 +1,336 @@
...
+            }
+            Err(err) => return Err(VolumeError::Inspect(err)),
```
Comment (harlem88) [inline edgehog-device-runtime-containers/src/docker/volume.rs:120]: same
```suggestion
            Err(err) => Err(VolumeError::Inspect(err)),
```

**Review by harlem88 — CHANGES_REQUESTED** [review]

File: edgehog-device-runtime-containers/src/docker/volume.rs (hunk 2 above)
Comment (joshuachp) [reply edgehog-device-runtime-containers/src/docker/volume.rs:118]: Thanks fixed

**Review by harlem88 — APPROVED** [review]

**Review by lucaato — APPROVED** [review]

**Pull request merged**